corehq/apps/users/models.py

```diff
@@ -180,6 +180,7 @@ class Permissions(DocumentSchema):
     download_reports = BooleanProperty(default=True)
     view_reports = BooleanProperty(default=False)
     view_report_list = StringListProperty(default=[])
+    edit_ucrs = BooleanProperty(default=False)
```

Comment:
Are you anticipating adding a migration for existing roles?

Reply:
On the product side, Kirti said that we should be having the default value `false` for every user (which is the case by default). And we are adding this change for SaaS environments only, so we would not be changing anything for solutions.

On the tech side, I asked Simon about it and he told me that 
> the SQL roles only store data for permissions that are 'True'

So we would not be requiring any migrations here.

Reply:
I think what @orangejenny is asking is whether or not we should be setting this to True for existing roles so that they don't loose access.

Reply:
I confirmed with Kirti that we don't have any active customers who are using UCRs at present. There is one customer who is currently in process of getting on-boarded and for them, we can create a role manually. 
So I don't think we would need to set it true for existing roles that have feature flags turned on.

Reply:
> And we are adding this change for SaaS environments only, so we would not be changing anything for solutions.

Isn't production used by both SaaS and solutions customers?

Reply:
> Isn't production used by both SaaS and solutions customers?

I was not aware of that. Thanks for bringing that up.
I guess in that case we would have to update the roles for the users having this FF turned on. I will talk to Kirti about it.

Reply:
I hate to backtrack on this, but now that I read the migration, it doesn't seem right that for users on the feature flag, the new permission is being added, so those users won't lose access, but also, any other users with the same role will gain access.

I don't know that there's a clean way to do this migration. But I also suspect this might just be an academic question - it looks like virtually all users of that flag are either superusers or personal email addresses of employees. I'm hoping that the others are all admins of their domains. Could you check what the actual impact of the migration would be?

Reply:
Thank you Jenny for raising these points. I was also thinking about them at starting and discussed the same with the product too. Here is why we decided to go with the approach.

> it doesn't seem right that for users on the feature flag, the new permission is being added, so those users won't lose access, but also, any other users with the same role will gain access.

I don't think that the other users will gain access to the UCRs because the Feature flag is not enabled for them. As per current checks if both the feature flag and permissions are enabled then the only user will have access to the UCR page.

> I don't know that there's a clean way to do this migration. But I also suspect this might just be an academic question - it looks like virtually all users of that flag are either superusers or personal email addresses of employees. I'm hoping that the others are all admins of their domains. Could you check what the actual impact of the migration would be?

Yes, you are correct. Most of them are superusers. I did a quick pass on production env and found out that there are 146 users with the Feature Flag enabled. These users are 3032 roles assigned out of which 2980 are admin roles, 49 non-admin roles and 3 somehow are part of roles that are not present in DB.
I have created a [spreadsheet](https://docs.google.com/spreadsheets/d/1uqj6jxyW1V3RORiP2biOLGrauyfeduzC7j9rJVGfE7s/edit#gid=0) of all roles that are non-admin and would be affected by this migration with their domains.

These are the stats for production but the current migration would make things easier for self-hosters too. They will not abruptly lose access to a feature that might be using before.

Reply:
> As per current checks if both the feature flag and permissions are enabled then the only user will have access to the UCR page.

Ah, I hadn't looked closely at the code recently. So there won't be any change in behavior for now. I do wonder what'll happen when we do remove the feature flag altogether (assuming that's the eventual plan), but I think this is fine for this current step. Thank you for the diligence and talking this through with product.